index.php

```diff
@@ -123,60 +80,11 @@
 $params = array_merge($params, $courseparams);
 $courselookup = $DB->get_records_sql($coursesql, $params);
 
-foreach ($cxsizes as $cxdata) {
-    $contextlevel = $cxdata->contextlevel;
-    $instanceid = $cxdata->instanceid;
-    $contextsize = $cxdata->filessize;
-    $contextbackupsize = (empty($cxdata->backupsize) ? 0 : $cxdata->backupsize);
-    if ($contextlevel == CONTEXT_USER) {
-        $usersizes[$instanceid] = $contextsize;
-        $userbackupsizes[$instanceid] = $contextbackupsize;
-        continue;
-    }
-    if ($contextlevel == CONTEXT_COURSE) {
-        $coursesizes[$instanceid] = $contextsize;
-        $coursebackupsizes[$instanceid] = $contextbackupsize;
-        continue;
-    }
-    if (($contextlevel == CONTEXT_SYSTEM) || ($contextlevel == CONTEXT_COURSECAT)) {
-        $systemsize = $contextsize;
-        $systembackupsize = $contextbackupsize;
-        continue;
-    }
-    // Not a course, user, system, category, see it it's something that should be listed under a course
-    // Modules & Blocks mostly.
-    $path = explode('/', $cxdata->path);
-    array_shift($path); // Get rid of the leading (empty) array item.
-    array_pop($path); // Trim the contextid of the current context itself.
-
-    $success = false; // Course not yet found.
-    // Look up through the parent contexts of this item until a course is found.
-    while (count($path)) {
-        $contextid = array_pop($path);
-        if (isset($courselookup[$contextid])) {
-            $success = true; // Course found.
-            // Record the files for the current context against the course.
-            $courseid = $courselookup[$contextid]->courseid;
-            if (!empty($coursesizes[$courseid])) {
-                $coursesizes[$courseid] += $contextsize;
-                $coursebackupsizes[$courseid] += $contextbackupsize;
-            } else {
-                $coursesizes[$courseid] = $contextsize;
-                $coursebackupsizes[$courseid] = $contextbackupsize;
-            }
-            break;
-        }
-    }
-    if (!$success) {
-        // Didn't find a course
-        // A module or block not under a course?
-        $systemsize += $contextsize;
-        $systembackupsize += $contextbackupsize;
-    }
-}
-$cxsizes->close();
-$sql = "SELECT c.id, c.shortname, c.category, ca.name FROM {course} c "
-       ."JOIN {course_categories} ca on c.category = ca.id".$extracoursesql;
+$sql = "SELECT c.id, c.shortname, c.category, ca.name, rc.filesize, rc.backupsize
+          FROM {course} c
+          JOIN {report_coursesize} rc on rc.course = c.id
+          JOIN {course_categories} ca on c.category = ca.id".$extracoursesql."
```

Comment:
minor quibble, we are using " so I think it would be more readable to put $extracoursesql on the next line still inside the string?

Reply:
yep - done.